### PR TITLE
Fix test failure on Mac about end-of-line

### PR DESCRIPTION
**Pull request by Sen-real**
##### Objective
A potential bug is discovered by running automation tests on Mac. Error "failed to read string on top level" is printed in console.

##### Abstractions
It's caused by an error in a line of code. Unlike Windows, when running tests on Mac, the calls to split and join methods (which was added for an edge case, and not needed with the given test data) will change the original string, and this is why the error is discovered. On Windows, the original string is untouched, and that original string is already in correct format, so the test case passes.

##### Tests performed
Tested on Mac. The error is gone. All 40 tests pass.

### CODE REVIEW

File: extension/src/project/readOnlyDocument.ts [hunk 1]
```diff
@@ -87,19 +87,19 @@ export class ReadonlyDocument implements vscode.TextDocument {
         return ret;
     }
 
-    initialize(fileContent: string, languageId: string) {
-        this.fileContent = fileContent.split('\r\n').join('\n').split('\n').join('\r\n');//to make sure every line ends with \r\n
+    initialize(rawContent: string, langId: string) {
+        this.fileContent = rawContent.split('\r\n').join('\n').split('\n').join('\r\n');//to make sure every line ends with \r\n
```
Comment (JD-Howard) [inline extension/src/project/readOnlyDocument.ts:91]: I've looked at this line several times before and always kind of wondered why we weren't doing something like this:
`rawContent.replace(/[^\r]\n/g, '\r\n')`

File: extension/src/project/readOnlyDocument.ts [hunk 2]
```diff
@@ -87,19 +87,19 @@ export class ReadonlyDocument implements vscode.TextDocument {
...
-        this.languageId = languageId;
+        this.languageId = langId;
 
         if (this.fileContent.length === 0) {
             this.lineCount = 0;
             this.lines = [];
         }
         else {
-            this.lines = fileContent.split('\r\n');
+            this.lines = this.fileContent.split('\r\n');
```
Comment (JD-Howard) [inline extension/src/project/readOnlyDocument.ts:102]: I tested this on my mochawesome PR branch by correcting this line. It ran all the tests and opened up in the browser perfectly after implementing your OS specific fix.

File: extension/src/project/readOnlyDocument.ts (hunk 1 above)
Comment (Sen-real) [reply extension/src/project/readOnlyDocument.ts]: Possibly because regular expressions are normally harder to maintain. I updated the code with your suggestion.

**Review by nigma456 — APPROVED** [review]
